src/services/healthcareProfessionalService.ts

```diff
@@ -1,13 +1,53 @@
-import { healthcareProfessionals } from '../mockData/mockData';
-import { HealthcareProfessional } from '../typeDefs/gqlTypes';
+import { HealthcareProfessional, LocaleName } from '../typeDefs/gqlTypes';
+import { HealthcareProfessional as PrismaHealthcareProfessional,
+    LocaleName as PrismaLocaleName } from '@prisma/client';
+
+import prisma from '../db/client';
+
+type HealthcareProfessionalAndRelations = (PrismaHealthcareProfessional & 
+    { names: PrismaLocaleName[]})
+
+function convertPrismaToGqlHealthcareProfessional(input: HealthcareProfessionalAndRelations) {
+    // TODO: populate the rest of the fields in a later PR
+    const ret = {
+        id: String(input.id),
+        names: Array<LocaleName>(),
+        specialties: [],
+        spokenLanguages: [],
+        acceptedInsuranceOptions: [],
+        degrees: []
+    } as HealthcareProfessional;
+
+    for (let i = 0; i < input.names.length; i++) {
+        ret.names.push(input.names[i] as LocaleName);
+    }
+
+    return ret;
+}
 
 // TODO: add a validation step for incoming parameters
-export const getHealthcareProfessionalById = (id: string) => {
-    const matchingProfessional = healthcareProfessionals.find(
-        (professional: HealthcareProfessional) => professional.id === id
-    );
+export const getHealthcareProfessionalById = async (id: string) => {
+    const found = await prisma.healthcareProfessional.findUnique({ where: {
+        id: parseInt(id)
+    }, 
+    include: {
+        names: true  
+    }});
 
-    return matchingProfessional;
+    return found ? convertPrismaToGqlHealthcareProfessional(found) : null;
 };
 
-export const getHealthcareProfessionals = () => healthcareProfessionals;
+export const getHealthcareProfessionals = async () => {
+    const healthPros = await prisma.healthcareProfessional.findMany({
+        include: {
+            names: true
+        }
+    });
+
+    const ret = Array<HealthcareProfessional>();
+
+    for (let i = 0; i < healthPros.length; i++) {
```

Comment:
actually, this is a perfect use case for `.map()`

```suggestion
    const gqlFormmatedEntities = healthPros.map(convertPrismaToGqlHealthcareProfessional)
```

Reply:
Not sure if I follow, and some of the refactor suggestions are conflicting. I am gonna push some of the changes and let you comment again.

Reply:
ok! I'll add an example to the new changes